pkg/pillar/cipher/handlecipher.go

```diff
@@ -203,10 +157,30 @@ func decryptCipherBlockWithECDH(ctx *DecryptCipherContext,
 	return []byte{}, errors.New("Unsupported Encryption protocol")
 }
 
+// validate the eve node certificate attributes
+// and return private Key, if any
+func lookupEveNodeCert(ctx *DecryptCipherContext,
+	status *types.CipherContextStatus) ([]byte, error) {
+
+	// validate the config
+	if len(status.DeviceCert) == 0 ||
+		status.HasError() {
+		return []byte{}, errors.New("invalid certificate")
+	}
+	// get the private key, for software Ecdh
+	sub := ctx.SubEveNodeCertConfig
```

Comment:
But this doesn't work if there is a tpm. Hence the functions which pass around pvtKey are all confusing; we just need to tell the actual decrypt code that it should use the device ECDH key, and get it from the TPM or read it from a file if there is no TPM. Hence no need to add a pvtKey argument AFAICT.